### PR TITLE
Console - Update Plan entities and start to connect PS component 

### PR DESCRIPTION
**Pull request by ThibaudAV**
## Issue

https://gravitee.atlassian.net/browse/APIM-1550

## Description

- Update plan entities with last change into management v2
- Start to connect gio-policy-studio input with api stuff

## Additional context

image

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-qubjfbvegm.chromatic.com)

### CODE REVIEW

**Review by loriepisicchio — APPROVED** [review]

File: gravitee-apim-console-webui/src/entities/management-api-v2/plan/updateBasePlan.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   * @example My Api plan
+   */
+  name?: string;
+  /**
+   * @description Simple order that could be used by a front end to display plans in a certain order. To highlight a plan on the portal for instance.
+   * @example 0
+   */
+  order?: number;
+  security?: {
+    configuration?: Record<string, never>;
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/entities/management-api-v2/plan/updateBasePlan.ts]: Why `never` and not `unknown`?

File: gravitee-apim-console-webui/src/management/api/policy-studio-v4/design/api-v4-policy-studio-design.component.ts [hunk 2]
```diff
@@ -25,11 +34,69 @@ import { Subject } from 'rxjs';
...
 
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  ngOnInit(): void {}
+        this.endpointsInfo = api.endpointGroups.flatMap((endpointGroup) =>
+          endpointGroup.endpoints.map((endpoint) => {
+            const endpointPlugin = endpoints.find((endpointPlugin) => endpointPlugin.id === endpoint.type);
+            return {
+              type: endpoint.type,
+              icon:
+                endpointPlugin && endpointPlugin.icon ? this.iconService.registerSvg(endpoint.type, endpointPlugin.icon) : 'gio:language',
```
Comment (jourdiw) [inline gravitee-apim-console-webui/src/management/api/policy-studio-v4/design/api-v4-policy-studio-design.component.ts:90]: Only endpoints need this added logic for `icon`?

**Review by jourdiw — APPROVED** [review]

File: gravitee-apim-console-webui/src/management/api/policy-studio-v4/design/api-v4-policy-studio-design.component.ts (hunk 2 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/management/api/policy-studio-v4/design/api-v4-policy-studio-design.component.ts:90]: hum right

File: gravitee-apim-console-webui/src/entities/management-api-v2/plan/updateBasePlan.ts (hunk 1 above)
Comment (ThibaudAV) [reply gravitee-apim-console-webui/src/entities/management-api-v2/plan/updateBasePlan.ts]: hum, bad copy past

**Pull request merged**